packages/eslint-plugin/src/rules/no-require-imports.ts

```diff
@@ -35,3 +39,28 @@ export default util.createRule({
     };
   },
 });
+
+/**
+ * Finds the variable by a given name in a given scope and its upper scopes.
+ * @param initScope A scope to start find.
+ * @param name A variable name to find.
+ * @returns A found variable or `null`.
+ */
+function getVariableByName(
```

Comment:
This is a copy-paste from https://github.com/typescript-eslint/typescript-eslint/blob/4a88de2c621ae645c070d995768006df8cb7e4df/packages/eslint-plugin/src/rules/no-shadow.ts#L329-L346